dm/syncer/ddl.go

```diff
@@ -192,6 +192,49 @@ func (s *Syncer) clearOnlineDDL(tctx *tcontext.Context, targetTable *filter.Tabl
 	return nil
 }
 
+func (s *Syncer) adjustTableCollation(ddlInfo *ddlInfo) {
+	if createStmt, ok := ddlInfo.originStmt.(*ast.CreateTableStmt); ok {
+		// create table like old table has no table_options
+		// See https://dev.mysql.com/doc/refman/5.7/en/create-table.html
+		if createStmt.ReferTable != nil {
+			fmt.Printf("RefTable")
+			return
+		}
+		for _, tableOption := range createStmt.Options {
+			fmt.Printf("Collation == %v \n", tableOption)
+			// already have 'Collation'
+			if tableOption.Tp == ast.TableOptionCollate {
+				return
+			}
+		}
+		sourceTable := ddlInfo.sourceTables[0]
+		collationInfo, err := utils.GetTableCollation(s.tctx.Ctx, s.fromDB.BaseDB.DB, sourceTable.Schema, sourceTable.Name)
+		if err != nil {
+			fmt.Printf("err == %v \n", err)
+			s.tctx.L().Warn("get source table collation info failed.", zap.Stringer("table", sourceTable), zap.Error(err))
```

Comment:
i perfer not mute this err, seems this err only happen when wo lost connection whit upstream ?

Reply:
ok, i will return this err.